src/WireMock.Net/Util/ProtoDefinitionHelper.cs

```diff
@@ -1,12 +1,50 @@
 // Copyright © WireMock.Net
 
+using System.Collections.Generic;
+using System.IO;
+using Stef.Validation;
 using WireMock.Models;
 using WireMock.Settings;
 
 namespace WireMock.Util;
 
-internal static class ProtoDefinitionHelper
+/// <summary>
+/// Some helper methods for Proto Definitions.
+/// </summary>
+public static class ProtoDefinitionHelper
 {
+    /// <summary>
+    /// Builds a dictionary of ProtoDefinitions from a directory.
+    /// - The key will be the filename without extension.
+    /// - The value will be the ProtoDefinition with an extra comment with the relative path to each <c>.proto</c> file so it can be used by the WireMockProtoFileResolver.
+    /// </summary>
+    /// <param name="directory">The directory to start from.</param>
+    public static ProtoDefinitionData FromDirectory(string directory)
+    {
+        Guard.NotNullOrEmpty(directory);
+
+        var dictionary = new Dictionary<string, string>();
+        var filePaths = Directory.EnumerateFiles(directory, "*.proto", SearchOption.AllDirectories);
+
+        foreach (var filePath in filePaths)
+        {
+            // Get the relative path to the directory (note that this will be OS specific).
+            var relativePath = PathUtils.GetRelativePath(directory, filePath);
+
+            // Build comment and get content from file.
+            var comment = $"// {relativePath}";
+            var content = File.ReadAllText(filePath);
+
+            // Only add the comment if it's not already defined.
+            var modifiedContent = !content.StartsWith(comment) ? $"{comment}\n{content}" : content;
+            var key = Path.GetFileNameWithoutExtension(filePath);
+
+            dictionary.Add(key, modifiedContent);
```

Comment:
I'm curious why the filename isn't used as the key here. That would make resolving imports work without the comment.
The only change needed is to replace `\` with `/` of the relative path.

Reply:
> I'm curious why the filename isn't used as the key here.
The filename is used: `var key = Path.GetFileNameWithoutExtension(filePath);`

> That would make resolving imports work without the comment.
The IProtoFileResolver is used in [ProtobufJsonConverterFileSystem.cs](https://github.com/StefH/ProtoBufJsonConverter/blob/main/src/ProtoBufJsonConverter/IO/ProtobufJsonConverterFileSystem.cs) which is required by protobuf-net to resolve additional files.

Reply:
From my debugging, it tries to resolve based on the import filename, which would match the key in the dictionary based on the relative file path, no?